### PR TITLE
Activity Location backend and is_cancelled Field

### PR DESCRIPTION
**Pull request by TeerapatTrepopsakulsin**
### Description

- #207 
- #208 

### Functional Acceptance Criteria

- Can create and edit activity to update attribute correctly as state in _Additional Information_

### Checklist

- [x] I have performed a self-review of my code.
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation, if applicable.
- [x] My code follows the Coding Guidelines.

### Type of Change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation update
- [ ] Other (please specify)

### Additional Information
- Don't forget to _migrate_
- `Locations`: _new_  model
  + `latitude`/`longitude`: DecimalField
  + Edit/Create activity with data `{'location': {'lat': <latitude>, 'lon':<longitude>}}` to set the value
  + location of the activity is `activity.locations`
- `Activity`
  - `on_site`: BooleanField, default is False
    + Edit/Create activity with data `{'on_site': bool}` to set the value
  - `is_cancelled`: BooleanField, default is False
    + True if the activity is cancelled
    + `is_active()` is False if `is_cancelled` is True
    + Edit activity with data `{'is_cancelled': bool}` to set the value
  - `locations`: Foreign key referencing `Locations` model

### CODE REVIEW

**Review by OmegaOoh — COMMENTED** [review]
Just want to let you know, there's more than one place I put the test data in. Another one is in `Edit Modal`

File: activities/models.py
Comment (OmegaOoh) [inline activities/models.py:1]: 2 Concerns here,
1. Why did you `from twisted.protocols.amp import Decimal` (also you didn't specify it in `requirements.txt` which you should
2. Location needed to not know about the activity. Activity is the one store's location.

File: activities/views/activity_detail.py
Comment (OmegaOoh) [inline activities/views/activity_detail.py:1]: Please remove dispensable comments.

**Review by OmegaOoh — CHANGES_REQUESTED** [review]
You need to wait for @Thanawas-Sirilertsathit for functionality of the apps but the codes most of it looks fine (not worth a wait though) and you need to address the concerns I write below.

**Review by Thanawas-Sirilertsathit — CHANGES_REQUESTED** [review]
Merge from main please so that I can know that location model is working or not

**Comment by Thanawas-Sirilertsathit** [comment]
Sorry for that, the previous activities don't work but newly created activities work well

**Review by Thanawas-Sirilertsathit — APPROVED** [review]
Functionality works well for newly created activities although it no longer supports the previous activities.

File: activities/models.py
Comment (TeerapatTrepopsakulsin) [reply activities/models.py:1]: 1) deleted
2) Modified
3) deleted

File: activities/views/util.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 """Utility module."""
-
+from email.policy import default
```
Comment (OmegaOoh) [inline activities/views/util.py:2]: What is this used for

**Review by OmegaOoh — CHANGES_REQUESTED** [review]

File: activities/views/util.py (hunk 1 above)
Comment (TeerapatTrepopsakulsin) [reply activities/views/util.py]: Accidentally added it.
Removed.

**Review by OmegaOoh — APPROVED** [review]

**Comment by Thanawas-Sirilertsathit** [comment]
Functionality checked, your work is compatible with the frontend well.

**Pull request merged**